lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -456,26 +457,35 @@ def confirm_cluster_reachable
     def confirm_namespace_exists
       st, err = nil
       with_retries(2) do
-        @namespace_info, err, st = kubectl.run("get", "namespace", @namespace, "-o", "json", use_namespace: false,
-           log_failure: true)
+        _, err, st = kubectl.run("get", "namespace", @namespace, use_namespace: false, log_failure: true)
         st.success? || err.include?(NOT_FOUND_ERROR)
       end
       raise FatalDeploymentError, "Failed to find namespace. #{err}" unless st.success?
       @logger.info("Namespace #{@namespace} found")
     end
 
+    def tags_from_namespace_labels
+      namespace_info = nil
+      with_retries(2) do
+        namespace_info, err, st = kubectl.run("get", "namespace", @namespace, "-o", "json", use_namespace: false,
+          log_failure: true)
```

Comment:
What is the point of duplicating the kubectl stuff?

Reply:
`confirm_namespace_exists` should not have unexpected side effects like getting the tags, let's separate each distinct bit of functionality into its own method.